src/main/java/space/obminyashka/items_exchange/service/impl/SendGridService.java

```diff
@@ -84,7 +83,18 @@ private static Personalization createPersonalizationAndSetParameters(EmailType e
             personalization.addDynamicTemplateData(key, getMessageSource(parameterSource));
         });
 
-        personalization.addDynamicTemplateData("url", host.concat(emailType.callbackEndpoint.replace("{code}", codeId.toString())));
+        String htmlAction = """
```

Comment:
Would you mind moving that to the class level (making a constant) and using it here for formatting?